File: Nível 1/2152 - Pepe, Já Tirei a Vela!/main.lua
quantia = tonumber(io.read())

for i = 1, quantia do 
    dados = io.read()
    hora, minuto, status = dados:match("(%S+)%s+(%S+)%s+(%S+)")

    if hora == "0" then
        hora = "00"
    elseif tonumber(hora) < 10 then
        hora = "0" .. hora 
    end

    if minuto == "0" then
        minuto = "00"
    elseif tonumber(minuto) < 10 then
        minuto = "0" .. minuto 
    end

    status = tonumber(status)

    if status == 1 then
        print(string.format("%s:%s - A porta abriu!", hora, minuto))
    elseif status == 0 then
        print(string.format("%s:%s - A porta fechou!", hora, minuto))
    end
end

File: Nível 1/2152 - Pepe, Já Tirei a Vela!/main.py
quantia = int(input("A: "))

for i in range(1, quantia + 1):
    hora, minuto, status = map(str, input("B: ").split(" "))

    if hora == "0":
        hora = "00"
    elif int(hora) < 10:
        hora = "0" + hora
    
    if minuto == "0":
        minuto = "00"
    elif int(minuto) < 10:
        minuto = "0" + minuto

    status = int(status)

    if status == 1:
        print(f"{hora}:{minuto} - A porta abriu!")
    elif status == 0:
        print(f"{hora}:{minuto} - A porta fechou!")